### PR TITLE
Add post overload for a specific `EventPriority`

### PR DESCRIPTION
**Pull request by Technici4n**

### CODE REVIEW

File: src/main/java/net/neoforged/bus/api/IEventBus.java [hunk 1]
```diff
@@ -202,5 +202,17 @@ public interface IEventBus {
      */
     <T extends Event> T post(T event);
 
+    /**
+     * Submit the event for dispatch to listeners registered with a specific {@link EventPriority}.
+     * <p>
+     * If the bus is not started yet, an exception will be thrown.
```
Comment (sciwhiz12) [inline src/main/java/net/neoforged/bus/api/IEventBus.java:208]: I screm at this not being in a `@throws`.

File: src/main/java/net/neoforged/bus/api/IEventBus.java [hunk 2]
```diff
@@ -202,5 +202,17 @@ public interface IEventBus {
...
+     * <p>
+     * Note that posting events phase-by-phase is overall a lot slower than a regular {@link #post(Event)} call.
```
Comment (sciwhiz12) [inline src/main/java/net/neoforged/bus/api/IEventBus.java:210]: Potential rephrase and addition:
```suggestion
     * Manually posting events phase-by-phase through this method is less performant than dispatching to all phases through a {@link #post(Event)} call. Prefer that method when per-phase dispatching is not needed.
```

File: bus-jmh/src/main/java/net/neoforged/bus/benchmarks/ManyBussesBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                busses[i].addListener(EventPriority.HIGH, TestEvent.class, event -> event.value++);
+            }
+            if (i % 9 == 0) {
+                busses[i].addListener(EventPriority.LOW, TestEvent.class, event -> event.value++);
+            }
+        }
+    }
+
+    @Benchmark
+    public int testManyBusses() {
```
Comment (sciwhiz12) [inline bus-jmh/src/main/java/net/neoforged/bus/benchmarks/ManyBussesBenchmark.java:37]: Could we potentially have another benchmark here which posts to each event via the `post(Event)` call, for comparison's sake?

Comment (Technici4n) [reply bus-jmh/src/main/java/net/neoforged/bus/benchmarks/ManyBussesBenchmark.java:37]: There you go:

```
Benchmark                                          Mode  Cnt      Score      Error  Units
ManyBussesBenchmark.testManyBusses                 avgt   15   1115.624 ▒   29.924  ns/op
ManyBussesBenchmark.testManyBussesPerPhase         avgt   15  16236.739 ▒ 1644.855  ns/op
```
(first one is `post(Event)`)

**Review by Matyrobbrt — APPROVED** [review]

**Pull request merged**